lib/Mongo/MongoClient.php

```diff
@@ -115,6 +117,12 @@ public function close($connection = null)
      */
     public function connect()
     {
+        try {
+            $this->forceConnect();
+        } catch (\MongoDB\Driver\Exception\Exception $e) {
+            ExceptionConverter::toLegacy($e);
+        }
```

Comment:
@alcaeus same here if you don't like it we can remove, I am doing this to make sure connect throw some exceptions but not sure if we should really enforce that.